### ISSUE TITLE
About the generation of graphic data

### ISSUE THREAD

**Issue by fjw525** [raghavian/graph_refinement#1]
Hello, can you tell me what method is used to turn the image data into a graphical representation?

**Comment by raghavian** [comment]
Image to graph representation is performed using the method in our [earlier work](https://arxiv.org/abs/1708.02096). The pipeline used for that was specific to the internal dataset and is unfortunately not publicly available.

**Comment by fjw525** [comment]
Thank you for your reply. I have another question. The graph data contains only over-tracked ‘nodes’ and ‘adj’?

**Comment by raghavian** [comment]
Yes. To pose the task as a refinement one, we include low probability airway candidates also. And irrespective of the confidence on the airway node, all nodes are over-connected to the nearest K-neighbours.

**Comment by fjw525** [comment]
Thank you for your reply. I encountered a problem when I was running the program. An error occurred when adj_flat was generated. Are your adj_dense and adj_target the same shape?And does your ‘nodes’ include x, y, z, radius and 10 directions? Similar to [x,y,z,r,vx,vy,vz.......].
Hope to get your reply

**Comment by raghavian** [comment]
Yes, the node features can be any meaningful attributes and we use the ones described in the paper. For small enough graphs you can work with dense adjacency matrices; however, for large ones you want to exploit their sparsity and it is well supported by sparse operations in Pytorch.

**Comment by fjw525** [comment]
Thank you for your reply. At present, I am having difficulty finding node features. Can you provide us with the code for generating graphs and node features? Looking forward to your reply

**Comment by raghavian** [comment]
The graph generation was performed on internal data and some of the choices might not translate well. However, as described in the paper it is based on Laplacian of Gaussians at multiple scales or any standard [blob detection method](https://scikit-image.org/docs/dev/auto_examples/features_detection/plot_blob.html). This can be then used to perform the Bayesian smoothing. Please read our manuscript which has all details [here](https://link.springer.com/chapter/10.1007/978-3-319-67675-3_6).

**Comment by fjw525** [comment]
Thank you for your reply, I have two other questions. 1. The article mentioned that the mean and variance of each node feature are generated by the ten connected nodes? 2. The ground truth is composed of idx_pos and idx_neg spliced together, will there be situations where the positions do not correspond? 
Looking forward to your reply

**Comment by raghavian** [comment]
Hi
1. The mean and variance are obtained based on the Bayesian smoothing of the tracks as described in [this work](https://link.springer.com/chapter/10.1007/978-3-319-67675-3_6)
2. By positions, do you mean the spatial location or the index in the adjacency matrix. This operation does not lose any indexing.

**Comment by puallee** [comment]
Did you successfully reproduce the generated graph data?
